<launch>
 
  <node pkg="tf" type="static_transform_publisher" name="base_to_link" args="0 0 0 0 0 0 base_footprint base_link 100" />
 	<node pkg="tf" type="static_transform_publisher" name="base_to_gyro" args="0 0 0 0 0 0 base_footprint gyro_link 100" />
  <node pkg="tf" type="static_transform_publisher" name="base_to_laser" args="0.15 0 0.05 3.14159 0 0 base_footprint laser 100" />

    <param name = "robot_description" textfile = "$(find huanyubot_description)/urdf/huanyubot.urdf"/>
    
    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />

	<!--Start the robot's base control code and publish the odometer data-->

	<node pkg="huanyu_robot_start" type="huanyu_robot_node" name="publish_odom" output="screen">
  	<param name="usart_port"         type="string" value="/dev/huanyu_base"/>  
  	<param name="baud_data"          type="int"    value="115200"/>
  	<param name="robot_frame_id"     type="string" value="base_footprint"/>
  	<param name="smoother_cmd_vel"   type="string" value="smoother_cmd_vel"/>
  	<param name="imu_velocity_z"     type="bool"   value="false"/>
  	<param name="filter_Vx_match"    type="double" value="1.0"/>
  	<param name="filter_Vth_match"   type="double" value="1.0"/>
	</node>

  <!--node pkg="hls_lfcd_lds_driver" type="hlds_laser_publisher" name="hlds_laser_publisher" output="screen">
    <param name="port" value="/dev/huanyu_laser"/>
    <param name="baud_rate" value="230400"/>
    <param name="frame_id" value="laser"/>
  </node-->

  <node name="rplidarNode"          pkg="rplidar_ros"  type="rplidarNode" output="screen">
    <param name="serial_port"         type="string" value="/dev/huanyu_laser"/>
    <param name="serial_baudrate"     type="int"    value="115200"/>
    <param name="frame_id"            type="string" value="laser"/>
    <param name="inverted"            type="bool"   value="false"/>
    <param name="angle_compensate"    type="bool"   value="true"/>
  </node>


  <!-- Robot pose ekf -->
  <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf" output="screen">
    <param name="freq" value="30.0"/>
    <param name="sensor_timeout" value="1.0"/>
    <param name="odom_used" value="true"/>
    <param name="imu_used" value="true"/>
    <param name="vo_used" value="false"/>
    <remap from="/imu_data" to="/mobile_base/sensors/imu_data" />
  </node>


  <arg name="node_name"             value="velocity_smoother"/>
  <arg name="nodelet_manager_name"  value="nodelet_manager"/>
  <arg name="config_file"           value="$(find huanyu_robot_start)/param/Huanyu_smoother.yaml"/>
  <arg name="raw_cmd_vel_topic"     value="cmd_vel"/>
  <arg name="smooth_cmd_vel_topic"  value="smoother_cmd_vel"/>
  <arg name="robot_cmd_vel_topic"   value="robot_cmd_vel"/>
  <arg name="odom_topic"            value="odom_combined"/>
  

  <node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager_name)" args="manager"/>
  
  <include file="$(find yocs_velocity_smoother)/launch/velocity_smoother.launch">
    <arg name="node_name"             value="$(arg node_name)"/>
    <arg name="nodelet_manager_name"  value="$(arg nodelet_manager_name)"/>
    <arg name="config_file"           value="$(arg config_file)"/>
    <arg name="raw_cmd_vel_topic"     value="$(arg raw_cmd_vel_topic)"/>
    <arg name="smooth_cmd_vel_topic"  value="$(arg smooth_cmd_vel_topic)"/>
    <arg name="robot_cmd_vel_topic"   value="$(arg robot_cmd_vel_topic)"/>
    <arg name="odom_topic"            value="$(arg odom_topic)"/>
  </include>

</launch>


